mypy/newsemanal/semanal.py

```diff
@@ -848,7 +847,12 @@ def visit_class_def(self, defn: ClassDef) -> None:
     def analyze_class(self, defn: ClassDef) -> None:
         fullname = self.qualified_name(defn.name)
         if not defn.info and not self.is_core_builtin_class(defn):
-            self.add_symbol(defn.name, PlaceholderNode(fullname, defn, True), defn)
+            # Add placeholder so that self-references in base classes can be
+            # resolved.  We don't want this to cause a deferral, since if there
+            # are no incomplete references, we'll replace this with a TypeInfo
+            # before returning.
+            self.add_symbol(defn.name, PlaceholderNode(fullname, defn, True), defn,
+                            can_defer=False)
```

Comment:
OK, this makes sense, in the test case `Iterable` will trigger `Name not defined` instead of deferring on final iteration. So we shouldn't defer here either.

I am just a bit worried that this may cause placeholder to leak to later stages, so I would add couple more tests, for example with a start import in the file where the class is defined.

Do we also need the same for `A = NewType('A', List['A'])`?

Reply:
This actually isn't required to fix the new test case. Without this, we do some extra iterations, and some other tests would fail. The test failures are likely caused by other bugs triggered by the extra pass. I'll create a separate issue about them.

I can add tests if you have specific ideas, but I'd rather not add test cases just in case, as they can give a false sense of security. A test case needs to do very particular things to trigger this error case, and small changes can turn it into something that doesn't test anything useful. The existing test case was a result of some trial an error.

`NewType` was a good catch -- it requires a similar treatment. I added a test case that relies on another bug, which is somewhat unfortunate. It would be better to have a test case that tests the number of semantic analysis iterations, but that's not possible right now.